### PR TITLE
Ensure CSS `theme()` functions are evaluated in media query ranges with collapsed whitespace

### LINKED ISSUE #14320: [v4] theme() function does not work with media query range syntax operators
**What version of Tailwind CSS are you using?**
`4.0.0-alpha.21`

**What build tool (or framework if it abstracts the build tool) are you using?**
Remix, Vite

**What version of Node.js are you using?**
`22.4.1`

**What browser are you using?**
Chrome, Safari and Firefox

**What operating system are you using?**
macOS Sonoma 14.6.1

**Reproduction URL**
[https://github.com/boylett/tailwindcss-v4-theme-in-media-bug](https://github.com/boylett/tailwindcss-v4-theme-in-media-bug)

**Describe your issue**
Using `theme()` in media queries does not work when using range syntax operators such as `<=` and `>=`.
See [root.css](https://github.com/boylett/tailwindcss-v4-theme-in-media-bug/blob/main/app/root.css) in the repro.

### PR DESCRIPTION
**Pull request by philipp-spiess**
Fixes #14320

This PR adds `>`, `<`, and `=` as separators into the CSS value parser. This is necessary because [`@media` range context](https://www.w3.org/TR/mediaqueries-4/#mq-range-context) does not require spaces around these operators so something like this is a valid value for the range syntax:

```css
@media (40rem<width<=48rem) { 
  /* ... */
}
```

If you add our CSS `theme()` function to the mix, this rule look like that:

```css
@media (theme(--breakpoint-sm)<width<=theme(--breakpoint-md)) {
  /* ... */
}
```

### CODE REVIEW

**Review by thecrypticace — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,7 +7,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-- Nothing yet!
+### Fixed
+
+- Correctly apply CSS `theme()` functions inside `@media` range contexts ((#14321)[https://github.com/tailwindlabs/tailwindcss/pull/14321])
```
Comment (adamwathan) [inline CHANGELOG.md:12]:
```suggestion
- Ensure CSS `theme()` functions are evaluated in media query ranges with collapsed whitespace ((#14321)[https://github.com/tailwindlabs/tailwindcss/pull/14321])
```

File: packages/tailwindcss/src/value-parser.ts [hunk 2]
```diff
@@ -157,7 +163,14 @@ export function parse(input: string) {
...
         for (; end < input.length; end++) {
           peekChar = input.charCodeAt(end)
-          if (peekChar !== COLON && peekChar !== COMMA && peekChar !== SPACE) {
+          if (
+            peekChar !== COLON &&
+            peekChar !== COMMA &&
+            peekChar !== SPACE &&
+            peekChar !== LESSER_THAN &&
+            peekChar !== GREAT_THAN &&
+            peekChar !== EQUAL
```
Comment (adamwathan) [inline packages/tailwindcss/src/value-parser.ts:172]:
```suggestion
            peekChar !== LESS_THAN &&
            peekChar !== GREATER_THAN &&
            peekChar !== EQUALS
```

File: packages/tailwindcss/src/value-parser.ts [hunk 3]
```diff
@@ -139,7 +142,10 @@ export function parse(input: string) {
       // ```
       case COLON:
       case COMMA:
-      case SPACE: {
+      case SPACE:
+      case LESSER_THAN:
+      case GREAT_THAN:
+      case EQUAL: {
```
Comment (adamwathan) [inline packages/tailwindcss/src/value-parser.ts:148]:
```suggestion
      case LESS_THAN:
      case GREATER_THAN:
      case EQUALS: {
```

File: packages/tailwindcss/src/value-parser.ts [hunk 4]
```diff
@@ -111,6 +111,9 @@ const DOUBLE_QUOTE = 0x22
 const OPEN_PAREN = 0x28
 const SINGLE_QUOTE = 0x27
 const SPACE = 0x20
+const LESSER_THAN = 0x3c
+const GREAT_THAN = 0x3e
+const EQUAL = 0x3d
```
Comment (adamwathan) [inline packages/tailwindcss/src/value-parser.ts:116]:
```suggestion
const LESS_THAN = 0x3c
const GREATER_THAN = 0x3e
const EQUALS = 0x3d
```

**Review by adamwathan — COMMENTED** [review]
Tweaked the changelog comment, updated Unicode constants to closer match the official character names:

https://www.unicode.org/charts/charindex.html

**Pull request merged**